dockstore-integration-testing/src/test/java/io/dockstore/webservice/ConfigurationIT.java

```diff
@@ -0,0 +1,131 @@
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+    @Rule
+    public TestRule watcher = new TestWatcher() {
+        protected void starting(Description description) {
+            System.out.println("Starting test: " + description.getMethodName());
+        }
+    };
+
+    private void before(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
+        CommonTestUtilities.dropAndRecreateNoTestData(support);
+        support.before();
+        testingPostgres = new TestingPostgres(support);
+        CommonTestUtilities.cleanStatePrivate2(support, false);
+    }
+
+    private void after(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
+        support.getEnvironment().healthChecks().shutdown();
+        support.after();
+        BaseIT.assertNoMetricsLeaks(support);
+    }
+
+    /**
+     * Create a Dropwizard test support object with the specified configuration overrides.
+     */
+    private DropwizardTestSupport<DockstoreWebserviceConfiguration> createSupport(ConfigOverride... overrides) {
+        return new DropwizardTestSupport<DockstoreWebserviceConfiguration>(DockstoreWebserviceApplication.class, DROPWIZARD_CONFIGURATION_FILE_PATH, overrides);
+    }
+
+    /**
+     * Run the specified Runnable in the environment corrsponding to the supplied Dropwizard test support object.
+     */
+    private void runWithSupport(DropwizardTestSupport<DockstoreWebserviceConfiguration> support, Runnable runnable) throws Exception {
+        before(support);
+        try {
+            runnable.run();
+        } finally {
+            after(support);
+        }
+    }
+
+    private long countSourceFilesWithPath(String path) {
+        // this isn't the most robust way to run a SELECT, but it's good enough for the purposes of this test.
+        return testingPostgres.runSelectStatement(String.format("select count(*) from sourcefile where path = '%s' or absolutepath = '%s'", path, path), long.class);
+    }
+
+    private void createWorkflow() {
+        ApiClient webClient = getWebClient(true, BaseIT.ADMIN_USERNAME, testingPostgres);
+        HostedApi api = new HostedApi(webClient);
+        Workflow hostedWorkflow = api.createHostedWorkflow("awesomeWorkflow", null, DescriptorLanguage.CWL.getShortName(), null, null);
+        SourceFile file = new SourceFile();
+        file.setContent("cwlVersion: v1.0\nclass: Workflow");
+        file.setType(SourceFile.TypeEnum.DOCKSTORE_CWL);
+        file.setPath("/Dockstore.cwl");
+        file.setAbsolutePath("/Dockstore.cwl");
+        SourceFile file2 = new SourceFile();
+        file2.setContent("{}");
+        file2.setType(SourceFile.TypeEnum.CWL_TEST_JSON);
+        file2.setPath(SOURCEFILE_PATH);
+        file2.setAbsolutePath(SOURCEFILE_PATH);
+        api.editHostedWorkflow(hostedWorkflow.getId(), List.of(file, file2));
+    }
+
+    @Test
```

Comment:
Thanks for the tip, Denis!  Since the last of these tests catches the exception, there's enough structural difference that using parameterized tests doesn't result in a big win in this case.   But I'll keep it in mind for future work!